### PR TITLE
PathExcludeGenerator: Add a bunch of patterns

### PR DESCRIPTION
**Pull request by fviernau**
These patterns have been identified by investigating the test assets of `PathExcludeGeneratorTest`.

### CODE REVIEW

File: helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt [hunk 1]
```diff
@@ -199,9 +201,82 @@ private val PATH_EXCLUDE_REASON_FOR_FILENAME = listOf(
...
+    "runsuite.c" to TEST_OF,
+    "runtest.c" to TEST_OF,
+    "settings.gradle" to BUILD_TOOL_OF,
+    "setup.cfg" to BUILD_TOOL_OF,
+    "setup.py" to BUILD_TOOL_OF,
+    "test_*.c" to BUILD_TOOL_OF
 ).checkNoDuplicatePatterns { it.first }
+
+fun main() {
+    PathExcludeGenerator.generatePathExcludes(listOf("a/b/c"))
```
Comment (tsteenbe) [inline helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt:275]: @fviernau  You adding a function here, not a pattern recommend to put this in a separate commit

**Review by tsteenbe — CHANGES_REQUESTED** [review]

File: helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt (hunk 1 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt:275]: This was left over from debugging.

**Review by tsteenbe — APPROVED** [review]

**Comment by fviernau** [comment]
Merged despite the unrelated test failures.
